streaming/__init__.py

```diff
@@ -13,3 +16,11 @@
     'StreamingDataLoader', 'StreamingDataset', 'CSVWriter', 'JSONWriter', 'MDSWriter', 'TSVWriter',
     'XSVWriter', 'LocalDataset', 'vision', 'text'
 ]
+
+IS_MACOS = sys.platform == 'darwin'
+
+# Set the multiprocessing start method to `fork` for MAC OS since
+# streaming uses a FileLock for sharing the resources between ranks
+# and workers which is Unpickleable except method `fork`.
+if IS_MACOS:
+    mp.set_start_method('fork', force=True)
```

Comment:
this should probably go instead dataset __init__?

Reply:
What if user uses `LocalDataset` ? Would they need this ?

Reply:
yeah, in both dataset `__init__`s (or a common dataset base class but I personally don't think there is enough to warrant a base class?)

Reply:
Done in https://github.com/mosaicml/streaming/pull/101/commits/b3e9ee30820f97457712ce7e529e12464add9cbe.